stm32-modules/include/flex-stacker/flex-stacker/motor_task.hpp

```diff
@@ -47,23 +48,61 @@ class MotorTask {
 
     template <MotorControlPolicy Policy>
     auto run_once(Policy& policy) -> void {
-        std::ignore = policy;
         if (!_task_registry) {
             return;
         }
 
         auto message = Message(std::monostate());
 
         _message_queue.recv(&message);
-        auto visit_helper = [this](auto& message) -> void {
-            this->visit_message(message);
+        auto visit_helper = [this, &policy](auto& message) -> void {
+            this->visit_message(message, policy);
         };
         std::visit(visit_helper, message);
     }
 
   private:
-    auto visit_message(const std::monostate& message) -> void {
-        static_cast<void>(message);
+    template <MotorControlPolicy Policy>
+    auto visit_message(const std::monostate& m, Policy& policy) -> void {
+        static_cast<void>(m);
+        static_cast<void>(policy);
+    }
+
+    template <MotorControlPolicy Policy>
+    auto visit_message(const messages::MotorEnableMessage& m, Policy& policy)
+        -> void {
+        auto response = messages::AcknowledgePrevious{.responding_to_id = m.id};
+        if (!(motor_enable(MotorID::MOTOR_X, m.x, policy) &&
+              motor_enable(MotorID::MOTOR_Z, m.z, policy) &&
+              motor_enable(MotorID::MOTOR_L, m.l, policy))) {
+            response.with_error = m.x ? errors::ErrorCode::MOTOR_ENABLE_FAILED
+                                      : errors::ErrorCode::MOTOR_DISABLE_FAILED;
+        };
+        static_cast<void>(_task_registry->send_to_address(
+            response, Queues::HostCommsAddress));
+    }
+
+    template <MotorControlPolicy Policy>
+    auto motor_enable(MotorID id, std::optional<bool> engage, Policy& policy)
+        -> bool {
+        if (!engage.has_value()) {
+            return true;
+        }
+        return engage ? policy.enable_motor(id) : policy.disable_motor(id);
```

Comment:
This will always return `true`, it should be `return engage.value() ? ...` instead.